packages/@expo/cli/src/export/exportStaticAsync.ts

```diff
@@ -90,13 +90,21 @@ export async function getFilesToExportFromServerAsync(
     // don't require the build-time generation of every possible group
     // variation.
     exportServer,
+    skipHtmlPrerendering,
     // name : contents
     files = new Map(),
   }: {
     manifest: ExpoRouterRuntimeManifest;
     serverManifest: RoutesManifest;
     renderAsync: (requestLocation: HtmlRequestLocation) => Promise<string>;
     exportServer?: boolean;
+    /**
+     * Skip HTML pre-rendering when SSR is enabled (HTML will be rendered at runtime).
+     *
+     * This is separate from `exportServer` because RSC mode also uses `exportServer: true`,
+     * but still needs placeholder HTML files.
+     */
+    skipHtmlPrerendering?: boolean;
```

Comment:
not important for this PR: Would love to (in theory) see this on the `ExpoRouterRuntimeManifest`, but we can get to that later. I think it'd be really neat if the manifest could drive what the `render.js` output is capable of and it also means that the manifest remains solely responsible for production routing

basically, let's hoist this if. possible to the manifest